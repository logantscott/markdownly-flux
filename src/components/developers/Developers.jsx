import React from 'react';
import styles from './Developers.css';

const Developers = () => {
  const developers = [
    {
      name: 'Nikki Kiga',
      title: 'Software Developer',
      image: 'nikki.jfif',
      github: 'nikki-kiga'
    },
    {
      name: 'Max Lamb',
      title: 'Software Developer',
      image: 'max.jfif',
      github: 'MaximusLamb'
    },
    {
      name: 'Logan Scott',
      title: 'Full-Stack Developer',
      image: 'logan.jfif',
      github: 'logantscott'
    }
  ];

  const devElements = developers.map(({ name, title, image, github }) => (
    <figure key={name}>
      <img alt={name} src={`..\\..\\public\\images\\${image}`} />
      <figcaption>
        <p className={styles.name}>{name}</p>
        <p>{title}</p>
        <a href={`https://github.com/${github}`}>Github</a>
      </figcaption>
    </figure>
  ));

  return (
    <section className={styles.Developers}>
      {devElements}
    </section>
  );
};

export default Developers;
