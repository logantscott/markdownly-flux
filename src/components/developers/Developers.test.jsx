import React from 'react';
import { shallow } from 'enzyme';
import Developers from './Developers';

describe('Developers component', () => {
  it('renders Developers', () => {

    const wrapper = shallow(
      <Developers />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
