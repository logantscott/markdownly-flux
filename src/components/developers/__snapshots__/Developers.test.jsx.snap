// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developers component renders Developers 1`] = `
<section
  className="Developers"
>
  <figure
    key="Nikki Kiga"
  >
    <img
      alt="Nikki Kiga"
      src="..\\\\..\\\\public\\\\images\\\\nikki.jfif"
    />
    <figcaption>
      <p
        className="name"
      >
        Nikki Kiga
      </p>
      <p>
        Software Developer
      </p>
      <a
        href="https://github.com/nikki-kiga"
      >
        Github
      </a>
    </figcaption>
  </figure>
  <figure
    key="Max Lamb"
  >
    <img
      alt="Max Lamb"
      src="..\\\\..\\\\public\\\\images\\\\max.jfif"
    />
    <figcaption>
      <p
        className="name"
      >
        Max Lamb
      </p>
      <p>
        Software Developer
      </p>
      <a
        href="https://github.com/MaximusLamb"
      >
        Github
      </a>
    </figcaption>
  </figure>
  <figure
    key="Logan Scott"
  >
    <img
      alt="Logan Scott"
      src="..\\\\..\\\\public\\\\images\\\\logan.jfif"
    />
    <figcaption>
      <p
        className="name"
      >
        Logan Scott
      </p>
      <p>
        Full-Stack Developer
      </p>
      <a
        href="https://github.com/logantscott"
      >
        Github
      </a>
    </figcaption>
  </figure>
</section>
`;
